src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -121,33 +122,42 @@ protected void Dispose(bool disposing)
 
                     // Eat any exceptions from the read-ahead task.  We don't need to log, as we expect
                     // failures from this task due to closing the connection while a read is in progress.
-                    if (_readAheadTask != null)
+                    ValueTask<int>? readAheadTask = ConsumeReadAheadTask();
+                    if (readAheadTask != null)
                     {
-                        ValueTask<int> t = _readAheadTask.GetValueOrDefault();
-                        if (t.IsCompleted)
-                        {
-                            if (!t.IsCompletedSuccessfully)
-                            {
-                                Exception ignored = t.AsTask().Exception; // accessing Exception prop is sufficient to suppress unobserved exception events
-                            }
-                        }
-                        else
-                        {
-                            t.AsTask().ContinueWith(p =>
-                            {
-                                Exception ignored = p.Exception;
-                            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnFaulted, TaskScheduler.Default);
-                        }
+                        IgnoreExceptionsAsync(readAheadTask.GetValueOrDefault());
                     }
                 }
             }
         }
 
+        /// <summary>Awaits a task, ignoring any resulting exceptions.</summary>
+        private static async void IgnoreExceptionsAsync(ValueTask<int> task)
```

Comment:
async void is the devil, but I know you know what you're doing...

Reply:
I agree on both counts.